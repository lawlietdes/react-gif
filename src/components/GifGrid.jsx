import { GifItem } from "./GifItem";
import { useFetchGrifs } from "../hooks/useFetchGrifs";

export const GifGrid = ( { category }) => {
    const { images, isLoading } = useFetchGrifs(category);

  return (
    <>
        <h3> {category} </h3> 
        { isLoading && <h2>Loading...</h2>}
        
        <div className = "card-grid">
          { images.map( (image) =>(
            <GifItem 
              key = { image.id } 
              { ...image }
            /> 
              
          ))}
        </div>
        
        
    </>
  )
}
