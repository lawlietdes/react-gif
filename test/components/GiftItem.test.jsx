import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GiftItem>', () => { 
    const title = 'This is a title';
    const url = 'https://media0.giphy.com/media/gk3R16JhLP8RUka2nD'; 
        
    test('debe hacer match con el Snapshot', () => {
        const { container }= render( <GifItem  title= { title } url={ url }/>);
        expect( container ).toMatchSnapshot();

     })
     test('debe mostrar el url indicado y el ALT indicado', () => { 
        render( <GifItem  title= { title } url={ url }/>);
        //screen.debug();
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
      })
      test('debe mostrar el titulo en el componente', () => { 
        render( <GifItem  title= { title } url={ url }/>);
        //screen.debug();
        expect( screen.getByText(title) ).toBeTruthy();
      })
 })