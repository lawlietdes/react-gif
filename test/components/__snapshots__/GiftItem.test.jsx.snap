// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem> debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="This is a title"
      src="https://media0.giphy.com/media/gk3R16JhLP8RUka2nD"
    />
    <p>
       
      This is a title
       
    </p>
  </div>
</div>
`;
